Code/Framework/AzToolsFramework/Tests/EditorViewportHelperTests.cpp

```diff
@@ -0,0 +1,247 @@
+            m_viewportSettings->m_helpersVisible = true;
+            m_viewportSettings->m_iconsVisible = true;
+
+            using ::testing::_;
+            using ::testing::Return;
+
+            EXPECT_CALL(*m_entityVisibleEntityDataCacheMock, GetVisibleEntityId(_))
+                .WillOnce(Return(m_entityId))
+                .WillOnce(Return(m_entityId2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, VisibleEntityDataCount()).WillByDefault(Return(2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityVisible(_)).WillByDefault(Return(true));
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            m_viewportSettings->Disconnect();
+            m_viewportSettings.reset();
+            m_editorHelpers.reset();
+            m_entityVisibleEntityDataCacheMock.reset();
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusDisconnect();
+        }
+
+        // EntityDebugDisplayEventBus overrides ...
+        // This function is called in DisplayComponents which is responsible for drawing the helpers, if this is called it means that a
+        // helper has been drawn
+        void DisplayEntityViewport(const AzFramework::ViewportInfo&, AzFramework::DebugDisplayRequests&) override
+        {       
+            const AZ::EntityId entityId = *AzFramework::EntityDebugDisplayEventBus::GetCurrentBusId();
+
+            m_displayEntityViewportEvent = true;
+            m_drawCount += 1;
+            m_drawnEntity = entityId;
+        }
+
+        AZ::EntityId m_entityId;
+        AZ::EntityId m_entityId2;
+        AZ::EntityId m_drawnEntity;
+        AzFramework::DebugDisplayRequests* m_debugDisplay = nullptr;
+        int m_drawCount = 0;
+        bool m_displayEntityViewportEvent = false;
+        AZStd::unique_ptr<ViewportSettingsTestImpl> m_viewportSettings;
+        AZStd::unique_ptr<AzToolsFramework::EditorHelpers> m_editorHelpers;
+        AZStd::unique_ptr<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>> m_entityVisibleEntityDataCacheMock;
+        AzFramework::CameraState m_cameraState;
+    };
+
+    TEST_F(EditorViewportOnlyDrawHelpersForSelectedEntityFixture, DisplayDebugDrawIfSelectedEntitiesOptionDisabledAndEntitySelected)
+    {
+        // Given the entity is selected and the option to only show helpers for selected entities is false
+        const AzToolsFramework::EntityIdList entityIds = { m_entityId };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+
+        m_viewportSettings->m_onlyShowForSelectedEntities = false;
+
+        // When the draw function is called
+        m_editorHelpers->DisplayHelpers(
+            AzFramework::ViewportInfo{ TestViewportId },
+            m_cameraState,
+            *m_debugDisplay,
+            [](AZ::EntityId)
+            {
+                return true;
+            });
```

Comment:
update: As this bit of code appears in every test it might be better to factor it out to a single function call that can either return true or false in the fixture...

```c++
EditorViewportOnlyDrawHelpersForSelectedEntityFixture::DisplayHelpers(bool visible)
{
m_editorHelpers->DisplayHelpers(
            AzFramework::ViewportInfo{ TestViewportId },
            m_cameraState,
            *m_debugDisplay,
            [visible](AZ::EntityId)
            {
                return visible;
            });
}

...

DisplayHelpers(true); // could then be called instead to reduce the amount of repetition
```